File: frontend/lib/api/zones.ts
import { apiFetch } from "@/lib/api";
import type { ZoneStatus } from "@/types/zone";

export async function getZoneStatus(includeSimulated: boolean = false): Promise<ZoneStatus[]> {
  const params = new URLSearchParams();
  if (includeSimulated) params.append("include_simulated", "true");
  return apiFetch<ZoneStatus[]>(`/zones/status?${params.toString()}`);
}

File: backend/app/routers/zones.py
from __future__ import annotations

from fastapi import APIRouter

from ..schemas.zone import ZoneStatusResponse
from ..services.zones import get_zone_status

router = APIRouter()

@router.get("/zones/status", response_model=list[ZoneStatusResponse])
def zone_status(include_simulated: bool = False) -> list[ZoneStatusResponse]:
    """Return the worst-case health category for each zone."""
    return get_zone_status(include_simulated)
